application/common/components/adapters/SagePeopleIdStore.php

```diff
@@ -0,0 +1,284 @@
+                $numItems,
+                var_export($employeeId, true)
+            ), 1558533849);
+        }
+    }
+
+    /**
+     * Get a list of users' information (containing at least an Employee ID) for
+     * all users changed since the specified time.
+     *
+     * @param int $unixTimestamp The time (as a UNIX timestamp).
+     * @return User[]
+     * @throws Exception
+     */
+    public function getUsersChangedSince(int $unixTimestamp)
+    {
+        throw new Exception(__FUNCTION__ . ' not yet implemented');
+    }
+
+    /**
+     * Get an access token by OAUTH request.
+     *
+     * @return string access token
+     * @throws Exception
+     */
+    private function getAccessToken(): string
+    {
+        $response = $this->getHttpClient()->post($this->authUrl, [
+            'connect_timeout' => $this->timeout,
+            'http_errors' => false,
+            'form_params' => [
+                'grant_type' => 'password',
+                'client_id' => $this->clientId,
+                'client_secret' => $this->clientSecret,
+                'username' => $this->username,
+                'password' => $this->password,
+            ],
+        ]);
+
+        $statusCode = (int)$response->getStatusCode();
+        if (($statusCode >= 200) && ($statusCode <= 299)) {
+            $data = Json::decode($response->getBody());
+            $accessToken = $data['access_token'] ?? '';
+        } else {
+            throw new Exception(sprintf(
+                'Unexpected response (%s %s): %s',
+                $response->getStatusCode(),
+                $response->getReasonPhrase(),
+                $response->getBody()
+            ), 1558380643);
+        }
+
+        return $accessToken;
+    }
+
+    /**
+     * Parse response body into a one-dimensional associative array of properties.
+     * @param array $body
+     * @return array
+     */
+    private function getRecordsFromResponseBody($body)
+    {
+        if ($body['done'] !== true) {
+            /*
+             * This flag will be false if the amount of data is too large to be returned by one
+             * query. At present, the maximum number of records returned in one response is 2000.
+             * https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_changing_batch_size.htm#!
+             * If the data set grows beyond this boundary, this adapter will need to be revised
+             * to support multi-batch queries.
+             * (https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_query.htm)
+             */
+            \Yii::error($this->getIdStoreName() . ' responded with incomplete data set');
+        }
+
+        return  array_map(
+            function ($item) {
+                $properties = Utils::arrayCollapseRecursive($item);
+                $properties[self::PROP_LOCKED] = $properties[self::PROP_LOCKED] ? 'yes' : 'no';
```

Comment:
Do you know whether these properties (PROP_LOCKED, PROP_REQUIRE_MFA) come through as a boolean or as a string? The User class has some extra logic for handling those various scenarios (`true` vs `'true'` vs `'yes'` vs `1` vs `'1'`), so I didn't know if this actually needed to be done here.

Reply:
They come in as boolean in the received json. Interesting. It wasn't working until I added this. Maybe there was a separate problem that was fixed at the same time. I'll take another look.